File: amqp/php/src/Ichnaea/Amqp/Xml/AgingWriter.php
<?php

namespace Ichnaea\Amqp\Xml;

use Ichnaea\Amqp\Model\Aging;

/**
 * An XML writer that writes Aging objects to xml
 *
 * @see Ichnaea\Amqp\Model\Aging
 * @author Miguel Ibero <miguel@ibero.me>
 */
class AgingWriter extends Writer
{
    /**
     * Constructor
     *
     * @param \DOMDocument the document to use
     * @param \DOMElement the element to use as root
     */    
    public function __construct($document=null, $root="aging")
    {
        parent::__construct($document, $root);
    }

    /**
     * Write the aging data into the xml
     *
     * @param Aging the aging object
     */
    public function build(Aging $aging)
    {
        $xmlRoot = $this->getRoot();
        foreach ($aging as $trial) {
            $xmlCol = $this->createElement("trial");
            $xmlRoot->appendChild($xmlCol);
            foreach ($trial as $key=>$value) {
                $xmlVal = $this->createElement("value", $value);
                $xmlVal->setAttribute("key", $key);
                $xmlCol->appendChild($xmlVal);
            }
        }
    }
}

File: amqp/src/main/java/edu/upc/ichnaea/amqp/data/AgingWriter.java
package edu.upc.ichnaea.amqp.data;

import java.io.IOException;
import java.io.Writer;

import edu.upc.ichnaea.amqp.model.Aging;
import edu.upc.ichnaea.amqp.model.AgingTrial;
import edu.upc.ichnaea.amqp.model.AgingValue;

public class AgingWriter {

    Writer mWriter;

    public AgingWriter(Writer writer) {
        mWriter = writer;
    }

    public AgingWriter write(Aging aging) throws IOException {

        for (AgingTrial trial : aging) {
            mWriter.write("\n");
            for (AgingValue value : trial) {
                mWriter.write(value.stringKey() + "\t" + value.stringValue()
                        + "\n");
            }
        }

        return this;
    }

    public void close() throws IOException {
        if (mWriter != null) {
            mWriter.close();
            mWriter = null;
        }
    }
}
